GooglePlayInstant/Tests/Editor/QuickDeploy/OAuth2ServerTest.cs

```diff
@@ -0,0 +1,110 @@
+            Assert.AreEqual(new KeyValuePair<string, string>("code", someString), codeResponse,
+                "Expected valid code response");
+
+            // response is error
+            var errorResponse =
+                OAuth2Server.GetAuthorizationResponse(new Uri(string.Format("{0}?error={1}", AddressPrefix,
+                    someString)));
+            Assert.AreEqual(new KeyValuePair<string, string>("error", someString), errorResponse,
+                "Expected invalid error response");
+
+            // response has invalid keys
+            var invalidResponse = new Uri(string.Format("{0}?someKey=someValue", AddressPrefix));
+            Assert.Throws<ArgumentException>(() => OAuth2Server.GetAuthorizationResponse(invalidResponse),
+                "Expected ArgumentException");
+            LogAssert.Expect(LogType.Error, OAuth2Server.InvalidQueryExceptionMessage);
+
+            // No response. Uri has no params
+            Assert.Throws<ArgumentException>(() => OAuth2Server.GetAuthorizationResponse(new Uri(AddressPrefix)),
+                "Expected ARgumentException");
+            LogAssert.Expect(LogType.Error, OAuth2Server.InvalidQueryExceptionMessage);
+        }
+
+        [Test]
+        public void TestUriContainsValidParams()
+        {
+            const string codeOrErrorKeyMustBePresentText =
+                "Uri query must include either \"code\" or \"error\" as keys.";
+
+            var validUriWithCode = new Uri(string.Format("{0}?code=someValue", AddressPrefix));
+            Assert.IsTrue(OAuth2Server.UriContainsValidQueryParams(validUriWithCode),
+                codeOrErrorKeyMustBePresentText);
+            var uriWithError = new Uri(string.Format("{0}?error=someValue", AddressPrefix));
+            Assert.IsTrue(OAuth2Server.UriContainsValidQueryParams(uriWithError),
+                codeOrErrorKeyMustBePresentText);
+
+            var invalidUriWithCodeAndError =
+                new Uri(string.Format("{0}?code=codeValue&error=errorValue", AddressPrefix));
+            Assert.IsFalse(OAuth2Server.UriContainsValidQueryParams(invalidUriWithCodeAndError),
+                "\"code\" and \"error\" cannot be present at the same time.");
+
+            var invalidUriWithOtherKeys =
+                new Uri(string.Format("{0}?code=codeValue&otherKey=someValue", AddressPrefix));
+            Assert.IsFalse(OAuth2Server.UriContainsValidQueryParams(invalidUriWithOtherKeys),
+                "No other keys apart from \"code\" and \"error\" are allowed.");
+
+            var invalidUriWithEmptyQuery = new Uri(AddressPrefix);
+            Assert.IsFalse(OAuth2Server.UriContainsValidQueryParams(invalidUriWithEmptyQuery),
+                "Uri with empty query should be invalid");
+        }
+
+        [Test]
+        public void TestGetQueryParamsFromUri()
+        {
+            // Case 1: Non-empty query
+            var testDictionary = new Dictionary<string, string> {{"a", "b"}, {"c", "d"}};
```

Comment:
Rename to expected or expectedDictionary.

Reply:
Resolved.